packages/api/cms-api/src/page-tree/document-subscriber.ts

```diff
@@ -0,0 +1,37 @@
+import { EntityManager, EntityName, EventArgs, EventSubscriber, MikroORM, UseRequestContext } from "@mikro-orm/core";
+import { Injectable, Type } from "@nestjs/common";
+
+import { DocumentInterface } from "../document/dto/document-interface";
+import { PageTreeService } from "./page-tree.service";
+
+export class DocumentSubscriberFactory {
+    static create({ Documents }: { Documents: Type<DocumentInterface>[] }): Type<EventSubscriber> {
+        @Injectable()
+        class DocumentSubscriber implements EventSubscriber<DocumentInterface> {
+            constructor(readonly em: EntityManager, private readonly pageTreeService: PageTreeService, private readonly orm: MikroORM) {
+                em.getEventManager().registerSubscriber(this);
+            }
+            getSubscribedEntities(): EntityName<DocumentInterface>[] {
+                return Documents;
+            }
+            async afterCreate(args: EventArgs<DocumentInterface>): Promise<void> {
+                await this.updatePageTreeNode(args.entity);
+            }
+            async afterUpdate(args: EventArgs<DocumentInterface>): Promise<void> {
+                await this.updatePageTreeNode(args.entity);
+            }
+
+            @UseRequestContext()
+            private async updatePageTreeNode(document: DocumentInterface): Promise<void> {
+                const pageTreeReadApi = this.pageTreeService.createReadApi({ visibility: "all" });
+                const pageTreeNode = await pageTreeReadApi.getFirstNodeByAttachedPageId(document.id);
+
+                if (pageTreeNode) {
+                    await pageTreeNode.assign({ ...pageTreeNode, updatedAt: new Date() });
+                    await this.em.flush();
```

Comment:
ok, good, now you don't flush anymore unexpectedly in a service method.

But do we really want to flush here at all? (the updatedAt change should be in the same transaction as the update itself)